wandb/sdk/artifacts/artifact_manifest_entry.py

```diff
@@ -199,6 +216,25 @@ def ref_url(self) -> str:
             + self.path
         )
 
+    def to_json(self) -> "DictEntry":
+        contents: DictEntry = {
+            "path": self.path,
+            "digest": self.digest,
+        }
+        if self.size is not None:
+            contents["size"] = self.size
+        if self.ref:
+            contents["ref"] = self.ref
+        if self.birth_artifact_id:
+            contents["birthArtifactID"] = self.birth_artifact_id
+        if self.local_path:
+            contents["local_path"] = self.local_path
+        if self.skip_cache:
+            contents["skip_cache"] = self.skip_cache
+        if self.extra:
+            contents["extra"] = self.extra
```

Comment:
comment(non-blocking): this is making me wonder if `ArtifactManifestEntry` would work as a `dataclass` (and e.g. use `asdict()` to handle some of the logic here).  curious if this has been considered before, but to be clear -- not a blocking request

Reply:
It would; perhaps we (I) should have documented our reasons, but I made it a dataclass in #4649 and then revered it to a non-dataclass in #5407 . IIRC it was too burdensome to handle the datatype conversions in the __post_init__, but we could consider it again